librz/core/cdebug.c

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_core.h>
+#include <rz_debug.h>
+
+RZ_API bool rz_core_debug_step_one(RzCore *core, int times) {
+	if (rz_config_get_i(core->config, "cfg.debug")) {
+		rz_reg_arena_swap(core->dbg->reg, true);
+		// sync registers for BSD PT_STEP/PT_CONT
+		rz_debug_reg_sync(core->dbg, RZ_REG_TYPE_GPR, false);
+		ut64 pc = rz_debug_reg_get(core->dbg, "PC");
+		rz_debug_trace_pc(core->dbg, pc);
+		if (!rz_debug_step(core->dbg, times)) {
+			eprintf("Step failed\n");
+			core->break_loop = true;
+			return false;
+		}
+	} else {
+		for (int i = 0; i < times; i++) {
+			rz_core_esil_step(core, UT64_MAX, NULL, NULL, false);
```

Comment:
also, the original `aes` command is also doing `.ar*`... maybe that's why tests fail

Reply:
I added back `rz_core_cmd0(core, ".ar*")` since it's better to refactor `ar` commands separately and replace with API calls it in an independent PR. There A LOT of `.ar*` calls across the Rizin codebase.